stdlib/asm/crypto/fri/helper.masm

```diff
@@ -30,41 +23,36 @@ export.generate_fri_parameters
     # load z from memory 
     padw
     exec.constants::z_ptr mem_loadw
-    #=> [(z1, z0)^n, z1, z0, lde_size, log2(lde_size), lde_g, 0, 0, ...] (6 cycles)
+    #=> [(z1, z0)^n, z1, z0, lde_size, log2(lde_size), lde_g, 0, ...] (6 cycles)
 
     # prepare stack
     drop
     drop
     dup.1
     dup.1
     dup.6
-    #=> [lde_g, z1, z0, z1, z0, lde_size, log2(lde_size), lde_g, 0, 0, ...] (5 cycles)
+    drop
+    #=> [z1, z0, z1, z0, lde_size, log2(lde_size), lde_g, 0, ...] (6 cycles)
 
-    # Compute trace generator `trace_g` = `lde_g^blowup_factor`
-    repeat.3
-        dup mul
-    end
-    #=> [trace_g, z1, z0, z1, z0, lde_size, log2(lde_size), lde_g, 0, 0, ...] (6 cycles)
+    # Load `trace_g` from memory
+    exec.constants::trace_domain_generator_ptr mem_load
+    #=> [trace_g, z1, z0, z1, z0, lde_size, log2(lde_size), lde_g, 0, ...] (2 cycles)
 
     # Compute `gz0` = `trace_g * g_0`
```

Comment:
this should be z_0 I believe, the same for z_1

Reply:
thanks!! fixed